### PR TITLE
Update `winapi` to 0.3

### PR DESCRIPTION
**Pull request by GabrielMajeri**
@retep998 [announced](https://github.com/retep998/winapi-rs/issues/316#issuecomment-343576108) that `winapi` 0.3 is feature complete, and that
> people should start porting and testing against the git version to ensure everything works before I finally publish winapi 0.3.

This pull request updates `winit` to use the new `winapi`, in preparation for its stable release.

### CODE REVIEW

**Comment by tomaka** [comment]
Thanks.

What's the strategy, though? Do we merge this in master in the hopes that `winapi` 0.3 gets released before a new version of winit is needed, or do we let this PR sit?

**Comment by GabrielMajeri** [comment]
@tomaka I will keep this pull request updated and wait for a new `winapi` version to be released.

I wrote this pull request to check if it is possible to update, resolve incompatibilities, etc.

**Comment by tomaka** [comment]
`winapi` 0.3 has been released. This needs a rebase/a merge and we're good to go.

File: Cargo.toml [hunk 1]
```diff
@@ -26,12 +26,19 @@ cocoa = "0.11"
...
 core-graphics = "0.10"
 
-[target.'cfg(target_os = "windows")'.dependencies]
-winapi = "0.2"
-shell32-sys = "0.1"
-user32-sys = "~0.1.2"
-kernel32-sys = "0.2"
-dwmapi-sys = "0.1"
+[target.'cfg(target_os = "windows")'.dependencies.winapi]
+git = "https://github.com/retep998/winapi-rs"
```
Comment (retep998) [inline Cargo.toml]: Don't forget to change this to `version = "0.3"`.

**Comment by GabrielMajeri** [comment]
@tomaka Is there anything else that has to be done before this pull request can be merged?

**Pull request merged**